runtime/json-functions.cpp

```diff
@@ -215,12 +215,41 @@ bool do_json_encode_string_vkext(const char *s, int len) noexcept {
 
 namespace impl_ {
 
+string JsonPath::to_string() const {
+  // this function is called only when error is occurred, so it's not
+  // very performance-sensitive
+
+  if (depth == 0) {
+    return string{"/", 1};
+  }
+  unsigned num_parts = std::clamp(depth, 0U, static_cast<unsigned>(arr.size()));
+  string result;
+  result.reserve_at_least((num_parts+1) * 8);
+  result.push_back('/');
+  for (unsigned i = 0; i < num_parts; i++) {
+    const char *key = arr[i];
+    if (key == nullptr) {
+      // int key indexing
+      result.append("[.]");
+    } else {
+      // string key indexing
+      result.append("['");
+      result.append(arr[i]);
+      result.append("']");
+    }
```

Comment:
Why did you choose the format
```
/['a'][.]['b']
```
instead of 
```
/a/#/b
```
?

Reply:
Because it mimics the way you would get that element. The chain of array index operations.
/-based paths are OK too, it they will be self-explanatory for people.
I opted for the format that is probably hard to be misinterpreted.